### DISCUSSION TITLE
Link evaded validation by beforeLoad

### DISCUSSION THREAD

**Discussion by 26king26** [TanStack/router#1693] [Q&A]
I am trying permission verification in beforeLoad, as it is a simple experiment, please do not worry about the name of my component.

I am attempting permission verification in the beforeLoad component of the root component, if it is not on the count page. Just check if there is a value for admin in localStorage, and if not, redirect to the count page. This is just a simulation of the login effect. Most of the results of the experiment met expectations, but I added a link button in the root component, which is roughly like this: root ->{link: {to: about}, outlet: {count}}.

At this moment, a magical thing happened. When I failed my first verification and redirected to the count page, when I clicked on the link again, it avoided the expected verification. My expectation is that whether I enter the URL in the search box or click on the link, it will go to the count page. There is no problem with the URL now, but when you click the button, it will go to count once, about again, and cross jump
![Snipaste_2024-06-02_00-16-25](https://github.com/TanStack/router/assets/84763745/1ed35599-ee11-4dce-8b9c-1f0b1355b458)
![Snipaste_2024-06-02_00-17-49](https://github.com/TanStack/router/assets/84763745/94c70e2d-32a3-40f1-94fb-ef838e7be80c)

**Comment by schiller-manuel** [comment]
can you please post a minimal complete example by forking one of the existing examples on stackblitz ?

**Reply by 26king26** [reply]
https://stackblitz.com/edit/github-flx8jx?file=src%2Froutes%2F__root.tsx

Should we do it this way? I'm not sure if you can see the cases on Stackblitz, haha, due to special reasons, I'm not proficient in using them.

If you can see the case, you will find that clicking the about button multiple times does not always complete the verification of the webpage, because I have already created a local Storage. setItem ('name ','not'); In theory, every verification cannot pass, and it should always point to the count page. But when I repeatedly click on the about link, it will go to count. When clicked again, it went to the about page again. Repeat like this. As expected, it should always point towards the count.

Because my React ability is just as bad as my English proficiency, haha, I don't know if it was supposed to be like this or if there was a problem with my usage somewhere.
